### ISSUE TITLE
A problem related to  Chromium CDP used by SipLoader

### ISSUE THREAD

**Issue by Web-Distortion** [SipLoader/SipLoader.github.io#7]
SipLoader uses the initiator of a request defined in Chromium CDP to determine dependencies among resources ([source code](https://github.com/SipLoader/SipLoader.github.io/blob/38cba7c42abb9110997e2aad31c45e8c2634afe5/dependency_base/dependency_tracker.js#L69) is here).
This is not fine-grained enough as we do not know when the requested resources are actually evaluated.

**Comment by SipLoader** [comment]
What's the exact meaning of  *fine-grained* here? Currently SipLoader evaluates a requested resource object when the object has no precedent dependencies.

**Comment by Web-Distortion** [comment]
*Fine-grained* here means that when a JS file A requests for another JS file B, SipLoader cannot determine when to evaluate B upon receiving it: should it 
1) pend the execution of A and directedly evaluate B, or
2) wait until A is entirely evaluated.

**Comment by SipLoader** [comment]
Yeah, that's the case. But tracking such fine-grained dependencies requires the data flow information during web page loading, which cannot be simply collected by Chromium DevTools Protocol. We are considering to integrate a finer-grained dependency tracking mechanism into SipLoader that can indicate the exact order of object evaluation.
